From d625ce06773bf0f0f0c94c7f2457d743b0de9a54 Mon Sep 17 00:00:00 2001
From: Fei Li <fei1.li@intel.com>
Date: Fri, 13 May 2022 10:05:51 +0800
Subject: [PATCH] misc: acrnd: fix a minor build Werror

The comparison ((argv + 1)) will always evaluate as 'true' for the pointer
operand in 'argv + 8' must not be NULL.

Tracked-On: #7453
Upstream-Status: Backported [https://github.com/projectacrn/acrn-hypervisor/commit/d625ce06773bf0f0f0c94c7f2457d743b0de9a54]
Signed-off-by: Fei Li <fei1.li@intel.com>
Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 misc/services/acrn_manager/acrnctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/misc/services/acrn_manager/acrnctl.c b/misc/services/acrn_manager/acrnctl.c
index 4a60442fe..2517c8d3f 100644
--- a/misc/services/acrn_manager/acrnctl.c
+++ b/misc/services/acrn_manager/acrnctl.c
@@ -657,7 +657,7 @@ static int valid_start_args(struct acrnctl_cmd *cmd, int argc, char *argv[])
 {
 	char df_opt[16] = "VM_NAME";
 
-	if (argc != 2 || ((argv + 1) && !strcmp(argv[1], "help"))) {
+	if (argc != 2 || !strcmp(argv[1], "help")) {
 		printf("acrnctl %s %s\n", cmd->cmd, df_opt);
 		return -1;
 	}
-- 
2.25.1

